nomad/structs/config/vault_test.go

```diff
@@ -66,69 +68,77 @@ func TestVaultConfig_Merge(t *testing.T) {
 func TestVaultConfig_IsEqual(t *testing.T) {
 	ci.Parallel(t)
 
-	require := require.New(t)
-
-	trueValue, falseValue := true, false
 	c1 := &VaultConfig{
-		Enabled:              &falseValue,
+		Enabled:              pointer.Of(false),
 		Token:                "1",
 		Role:                 "1",
-		AllowUnauthenticated: &trueValue,
+		Namespace:            "1",
+		AllowUnauthenticated: pointer.Of(true),
 		TaskTokenTTL:         "1",
 		Addr:                 "1",
+		ConnectionRetryIntv:  time.Second,
 		TLSCaFile:            "1",
 		TLSCaPath:            "1",
 		TLSCertFile:          "1",
 		TLSKeyFile:           "1",
-		TLSSkipVerify:        &trueValue,
+		TLSSkipVerify:        pointer.Of(true),
 		TLSServerName:        "1",
 	}
 
 	c2 := &VaultConfig{
-		Enabled:              &falseValue,
+		Enabled:              pointer.Of(false),
 		Token:                "1",
 		Role:                 "1",
-		AllowUnauthenticated: &trueValue,
+		Namespace:            "1",
+		AllowUnauthenticated: pointer.Of(true),
 		TaskTokenTTL:         "1",
 		Addr:                 "1",
+		ConnectionRetryIntv:  time.Second,
 		TLSCaFile:            "1",
 		TLSCaPath:            "1",
 		TLSCertFile:          "1",
 		TLSKeyFile:           "1",
-		TLSSkipVerify:        &trueValue,
+		TLSSkipVerify:        pointer.Of(true),
 		TLSServerName:        "1",
 	}
 
-	require.True(c1.IsEqual(c2))
+	// TODO: must.Equals(t, c, c2) should work here?
+	require.True(t, c.Equals(c2))
```

Comment:
@shoenig this seems like a possible bug in `test`?

Reply:
i believe the bug is the pointer comparison of TLSSkipVerify

````
	if c.TLSSkipVerify != b.TLSSkipVerify {
		return false
	}
```

Reply:
In fact if you fmt.Println(c1.Equals(c2)) it's false; testify is doing bad magic here to make it work

Reply:
Oh ho! Thanks!

Reply:
Done in [3f63499](https://github.com/hashicorp/nomad/pull/14298/commits/3f6349922eab6c6c77ec925cf36095ffeb4f5aa1)